src/common/module_builder.cc

```diff
@@ -65,10 +66,17 @@ ModuleBuilder::ModuleBuilder()
   m_context->appendDialectRegistry(registry);
 }
 
+bool ModuleBuilder::isOutputScalar(int index) const {
+  assert(index < m_is_output_scalar.size() && "Index of output out of range.");
```

Comment:
Nit: Change to `Output index is out of range`. Also assert/error messages shouldn't be terminated with a dot.